### PR TITLE
onBehalf logic added

### PR DESCRIPTION
**Pull request by clementbalestrat**
Screen Shot 2021-07-07 at 5 31 39 pm
Screen Shot 2021-07-07 at 5 31 13 pm

### CODE REVIEW

**Review by JChiaramonte7 — APPROVED** [review]
LGTM

File: queries/delegate/useGetDelegateWallets.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+const useGetDelegateWallets = (options?: QueryConfig<[AuthoriserWallet]>) => {
+	const isAppReady = useRecoilValue(appReadyState);
+	const network = useRecoilValue(networkState);
+	const walletAddress = useRecoilValue(walletAddressState);
+	const isL2 = useRecoilValue(isL2State);
+
+	return useQuery<[AuthoriserWallet]>(
+		QUERY_KEYS.Delegate.AuthorisedWallets(walletAddress ?? '', network?.id!),
+		async () => {
+			const { delegatedWallets } = await request(
```
Comment (andytcf) [inline queries/delegate/useGetDelegateWallets.ts:22]: haha @clementbalestrat, have we ditched the old  pageResults now

File: queries/delegate/useGetDelegateWallets.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+				}) => ({
+					address: authoriser,
+					canMint,
+					canBurn,
+					canClaim,
+				})
+			);
+		},
+		{
+			enabled: isAppReady && !isL2,
```
Comment (andytcf) [inline queries/delegate/useGetDelegateWallets.ts:58]: @clementbalestrat might want to check for ```walletAddress !== ''```

Comment (andytcf) [reply queries/delegate/useGetDelegateWallets.ts:58]: or ```isWalletConnected```

File: sections/shared/modals/WalletOptionsModal/WalletOptionsModal.tsx [hunk 3]
```diff
@@ -195,11 +205,32 @@ const WalletOptionsModal: FC<WalletOptionsProps> = ({ onDismiss, setWatchWalletM
...
+									onDismiss();
+									setDelegateModalOpened(true);
+								}}
+							>
+								{delegateIcon} {t('modals.wallet.delegate-mode.menu-title')}
+							</StyledButton>
+						)}
 					</Buttons>
 					<StyledDivider />
+					{delegateWallet && (
```
Comment (andytcf) [inline sections/shared/modals/WalletOptionsModal/WalletOptionsModal.tsx:224]: We should consider making the text have a defined width so the icons line up 

image

File: sections/staking/components/BurnTab/BurnTab.tsx [hunk 4]
```diff
@@ -217,24 +240,36 @@ const BurnTab: React.FC = () => {
...
 						});
+						transaction = delegateWallet
+							? await Synthetix[burnFunc](delegateWallet.address, amountToBurnBN, {
+									gasPrice: normalizedGasPrice(gasPrice),
+									gasLimit,
+							  })
+							: await Synthetix[burnFunc](amountToBurnBN, {
+									gasPrice: normalizedGasPrice(gasPrice),
+									gasLimit,
+							  });
```
Comment (andytcf) [inline sections/staking/components/BurnTab/BurnTab.tsx:272]: L: 291 missing delegateWallet in useCallback dependencies

File: sections/staking/components/StakingInput/StakingInput.tsx [hunk 5]
```diff
@@ -102,6 +102,7 @@ const StakingInput: React.FC<StakingInputProps> = ({
 	} = useStakingCalculations();
 	const { connectWallet } = Connector.useContainer();
 	const isWalletConnected = useRecoilValue(isWalletConnectedState);
+	const delegateWallet = useRecoilValue(delegateWalletState);
```
Comment (andytcf) [inline sections/staking/components/StakingInput/StakingInput.tsx:105]: this ```delegateWallet``` is not used

File: translations/en.json [hunk 6]
```diff
@@ -98,9 +98,16 @@
 			"etherscan": "View on Etherscan",
 			"disconnect-wallet": "disconnect",
 			"stop-watching": "Stop watching",
+			"stop-delegation": "Stop delegation",
```
Comment (andytcf) [inline translations/en.json:101]: i think it should be "stop delegate mode"

**Review by andytcf — CHANGES_REQUESTED** [review]
Left some fixes as comments

**Comment by andytcf** [comment]
@clementbalestrat 

small change about off-centered 

image

**Comment by andytcf** [comment]
# Checklist 

Flows tested

- [ ] With "all" delegation given, the delegator is able to mint custom
- [ ] NOTE: Revoked "all" delegation from the delegator, when logged in as a delegator, the wallet still appears in the delegate select modal
- [ ] With no permissions granted, minting shows the correct error, burning shows the correct error and claiming shows the correct error


Flows not tested due to time

- [ ] Burning works

**Comment by drptbl** [comment]
Issues:
-  Revoked "all" delegation from the delegator, when logged in as a delegator, the wallet still appears in the delegate select modal
- There is no way to remove a wallet completely from delegation list after you add it (you can only revoke it's delegations, but not remove it completely from the list)
Screenshot 2021-07-15 at 00 48 50
- Unhandled errors after going to "Burn max" right after minting
Screenshot_2021-07-15_at_01 25 09
- UI is glitched because of very long error message in the button on "Burn custom":
Screenshot_2021-07-15_at_01 26 47

**Review by vbstreetz — APPROVED** [review]
Looking good 👍🏽. Would be cool to retain kovan support though.

**Comment by clementbalestrat** [comment]
> Looking good 👍🏽. Would be cool to retain kovan support though.

Yeah we'll put it back once we have a testnet support with the Graph

**Comment by vbstreetz** [comment]
> > Looking good 👍🏽. Would be cool to retain kovan support though.
> 
> Yeah we'll put it back once we have a testnet support with the Graph

Awesome 👍🏽

**Comment by vbstreetz** [comment]
> > > Looking good 👍🏽. Would be cool to retain kovan support though.
> > 
> > 
> > Yeah we'll put it back once we have a testnet support with the Graph
> 
> Awesome 👍🏽

Wait, what did you mean exactly? since thegraph do support kovan etc...

**Comment by clementbalestrat** [comment]
> > > > Looking good 👍🏽. Would be cool to retain kovan support though.
> > > 
> > > 
> > > Yeah we'll put it back once we have a testnet support with the Graph
> > 
> > 
> > Awesome 👍🏽
> 
> Wait, what did you mean exactly? since thegraph do support kovan etc...

Yes but synthetix subgraphes are only on Mainnet for now.

**Comment by drptbl** [comment]
Everything has been fixed,

1 more issue:
- it would be great to redirect an user away from a page which he can't access while in delegation mode

**Review by drptbl — APPROVED** [review]

**Pull request merged**